### PR TITLE
PP-940 Create InquiryGatewayRequest and refactor inquire()

### PR DESCRIPTION
**Pull request by simad**
## WHAT
- `inquire()` now follows the same pattern as the other PaymentProvider methods
- transactionId and gatewayAccount are embedded in the InquiryGatewayRequest
  ## HOW 
  _Steps to test or reproduce:_

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/model/InquiryGatewayRequest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.pay.connector.model;
+
+import uk.gov.pay.connector.model.domain.ChargeEntity;
+import uk.gov.pay.connector.model.domain.GatewayAccountEntity;
+
+public class InquiryGatewayRequest implements GatewayRequest {
+
+    private final GatewayAccountEntity gatewayAccountEntity;
```
Comment (ckalista) [inline src/main/java/uk/gov/pay/connector/model/InquiryGatewayRequest.java]: Not so sure we should tie the whole entity with the Request. Can we no just keep what we need from the entity instead?

File: src/main/java/uk/gov/pay/connector/service/worldpay/WorldpayPaymentProvider.java [hunk 2]
```diff
@@ -74,32 +74,14 @@ public GatewayResponse cancel(CancelGatewayRequest request) {
...
-    private GatewayResponse mapToInquiryResponse(GatewayClient.Response response) {
-        return reduce(
-                client.unmarshallResponse(response, WorldpayOrderStatusResponse.class)
-                        .bimap(
-                                GatewayResponse::with,
-                                GatewayResponse::with
-
-                        )
-        );
+    public GatewayResponse inquire(InquiryGatewayRequest request) {
```
Comment (ckalista) [inline src/main/java/uk/gov/pay/connector/service/worldpay/WorldpayPaymentProvider.java]: yeah 👍

**Comment by ckalista** [comment]
Code review completed - just one comment.

File: src/main/java/uk/gov/pay/connector/model/InquiryGatewayRequest.java (hunk 1 above)
Comment (simad) [inline src/main/java/uk/gov/pay/connector/model/InquiryGatewayRequest.java]: In theory we don't need the whole object, true, but we have to keep it to conform to the GatewayRequest interface, which has `GatewayAccountEntity getGatewayAccount();`

**Pull request merged**